scapy/arch/linux.py

```diff
@@ -461,12 +461,18 @@ def __init__(self, iface=None, type=ETH_P_ALL, promisc=None, filter=None,
         )
         if not six.PY2:
             # Receive Auxiliary Data (VLAN tags)
-            self.ins.setsockopt(SOL_PACKET, PACKET_AUXDATA, 1)
-            self.ins.setsockopt(
-                socket.SOL_SOCKET,
-                SO_TIMESTAMPNS,
-                1
-            )
+            try:
+                self.ins.setsockopt(SOL_PACKET, PACKET_AUXDATA, 1)
+                self.ins.setsockopt(
+                    socket.SOL_SOCKET,
+                    SO_TIMESTAMPNS,
+                    1
+                )
+                self.auxdata_available = True
+            except OSError:
+                # Note: Auxiliary Data is only supported since
+                #       Linux 2.6.21
+                warning("Your Linux Kernel does not support Auxiliary Data!")
```

Comment:
this sounds a little too aggressive, since users can't disable it

Reply:
Do you want me to change this to an error instead of a warning?

Reply:
I was thinking ignore it silently ? or use a lighter logging like `log_runtime.info` so that it isn't displayed when importing scapy as a library.

Unlike other warnings users can't really do anything to avoid this from being triggered, so they would just disable all logs instead

Reply:
Done!